eksupgrade/src/k8s_client.py

```diff
@@ -424,80 +408,37 @@ def update_addons(cluster_name, version, vpcPass, regionName):
-            elif "aws-node" in pod.metadata.name and image.split(":")[-1] != "v" + cni_new and not vpcPass:
-                print(pod.metadata.name, "Current Version = ", image.split(":")[-1], "Updating To = ", "v" + cni_new)
+                queue.put([cluster_name, "kube-system", new_pod_name, "cluster-autoscaler", region_name])
+            elif "aws-node" in pod.metadata.name and _current_image != "v" + cni_new and not vpc_pass:
+                print(pod.metadata.name, "Current Version = ", _current_image, "Updating To = ", "v" + cni_new)
                 if flag_vpc:
-                    # TODO: Make this load safe.  This file is never closed. Use context manager instead.
-                    vpc_cni_yaml = open("eksupgrade/src/S3Files/vpc-cni.yaml", "r")
-                    body = yaml.safe_load(vpc_cni_yaml)
-                    body["spec"]["template"]["spec"]["containers"][0]["image"] = image.split(":")[0] + ":v" + cni_new
+                    with open("eksupgrade/src/S3Files/vpc-cni.yaml", "r", encoding="utf-8") as vpc_cni_yaml:
+                        body = yaml.safe_load(vpc_cni_yaml)
+
+                    body["spec"]["template"]["spec"]["containers"][0]["image"] = image_base_uri + ":v" + cni_new
                     old = body["spec"]["template"]["spec"]["initContainers"][0]["image"]
                     body["spec"]["template"]["spec"]["initContainers"][0]["image"] = old.split(":")[0] + ":v" + cni_new
-                    api_response = api_instance.patch_namespaced_daemon_set(
+                    apps_v1_api.patch_namespaced_daemon_set(
                         namespace="kube-system", name="aws-node", body=body, pretty=True
                     )
                     flag_vpc = False
                 time.sleep(20)
                 new_pod_name = sort_pods(
                     cluster_name=cluster_name,
-                    regionName=regionName,
+                    regionName=region_name,
                     original_name=pod.metadata.name,
                     old_pods_name=old_pods_name,
                     pod_name="aws-node",
                     nameSpace="kube-system",
                 )
-                print("old vpc cni Pod {oldp} \t new vpc cni {newp}".format(oldp=pod.metadata.name, newp=new_pod_name))
-                queue.put([cluster_name, "kube-system", new_pod_name, "aws-node", regionName])
+                print(f"Old VPC CNI pod: {pod.metadata.name} \t New VPC CNI pod: {new_pod_name}")
+                queue.put([cluster_name, "kube-system", new_pod_name, "aws-node", region_name])
         queue.join()
-        # if addon_status(cluster_name=cluster_name,nameSpace='kube-system',new_pod_name=new_pod_name, podName="aws-node",regionName=regionName) != True:
-        #     raise Exception("Pod is Not Started"+pod.metadata.name)
-
     except Exception as e:
         print(e)
         raise Exception(e)
 
 
-# print(list_nodes())
-# print(update_addons())
-def list_pods(ClusterName):
```

Comment:
Simply removed this because it's not in use anywhere

Reply:
thank goodness - can be a VERY expensive operation